src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -792,6 +792,13 @@ inline jlong min_signed_integer(BasicType bt) {
   return min_jlong;
 }
 
+inline uint bits_per_java_integer(BasicType bt) {
+  if (bt == T_INT) {
+    return BitsPerJavaInteger;
+  }
+  return BitsPerJavaLong;
```

Comment:
I think it'd be nice to add `assert(bt == T_LONG, "unsupported");` before the last return, like in the helper methods above.

Reply:
Indeed. Added in new commit.